thirdparty/faiss/faiss/IndexIVFPQ.cpp

```diff
@@ -418,9 +418,10 @@ void initialize_IVFPQ_precomputed_table(
     // squared norms of the PQ centroids
     std::vector<float> r_norms(pq.M * pq.ksub, NAN);
     for (int m = 0; m < pq.M; m++)
-        for (int j = 0; j < pq.ksub; j++)
-            r_norms[m * pq.ksub + j] =
-                    fvec_norm_L2sqr(pq.get_centroids(m, j), pq.dsub);
+        for (int j = 0; j < pq.ksub; j++) {
+            float norm = fvec_norm_L2sqr(pq.get_centroids(m, j), pq.dsub);
+            r_norms[m * pq.ksub + j] = (norm == 0.0 ? 1.0 : norm);
```

Comment:
no, because it uses norm for distance = x^2 - 2xy + y^2 calculation, not for a cosine-related code